core/errors.ml

```diff
@@ -10,11 +10,18 @@ type synerrspec = {filename : string; linespec : string;
 exception UndefinedVariable of string
 
 exception Type_error of (Position.t * string)
-exception MultiplyDefinedToplevelNames of ((Position.t list) stringmap)
+exception MultiplyDefinedMutualNames of ((Position.t list) stringmap)
+exception InvalidMutualBinding of Position.t
 exception RichSyntaxError of synerrspec
 exception SugarError of (Position.t * string)
 exception Runtime_error of string
 exception UnboundTyCon of (Position.t * string)
+exception InternalError of { filename: string; message: string }
+exception TypeApplicationArityMismatch of
+  { pos: Position.t; name: string; expected: int; provided: int}
+exception TypeApplicationKindMismatch of
```

Comment:
Oh, a record. This is nice, though inconsistent with other errors that store just tuples. Would it be a good idea to refactor other errors to also store records `{ pos: Position.t; message: string }`?

Reply:
I think so, yes. This points to a bigger issue, actually -- inline records are (relatively) recent in OCaml, and I think a lot of the nested tuples (for example, throughout the AST) may benefit from being redefined as inline records to make the source more descriptive.